crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -703,6 +706,50 @@ impl From<SyncTimelineEventDeserializationHelperV0> for SyncTimelineEvent {
     }
 }
 
+/// Represent a failed to send error of an event sent via the send_queue.
+/// These errors are not automatically retrievable, but yet some manual action
+/// can be taken before retry sending.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Enum))]
+pub enum QueueWedgeError {
+    /// This error occurs when there are some insecure devices in the room, and
+    /// the current encryption setting prohibit sharing with them.
+    InsecureDevices {
+        /// The insecure devices as a Map of userID to deviceID.
+        user_device_map: HashMap<String, Vec<String>>,
```

Comment:
Outside the FFI layer, we're not using stringly-typing, so this would become a `HashMap<OwnedUserId, Vec<OwnedDeviceId>>`, and similarly below.

Reply:
see https://github.com/matrix-org/matrix-rust-sdk/pull/4137/commits/3e657538a61fd298255504f75b9b2c20fdd75b82